template/src/ontology/Makefile.jinja2

```diff
@@ -342,6 +341,26 @@ $(REPORTDIR)/$(SRC)-obo-report.tsv: $(SRCMERGED) | $(REPORTDIR)
 
 $(REPORTDIR)/%-obo-report.tsv: % | $(REPORTDIR)
 	$(ROBOT) report -i $< $(REPORT_LABEL) $(REPORT_PROFILE_OPTS) --fail-on $(REPORT_FAIL_ON) {% if project.robot_report.use_base_iris %}{% if project.namespaces is not none %}{% for iri in project.namespaces %}--base-iri {{ iri }} {% endfor %}{% else %}--base-iri $(URIBASE)/{{ project.id.upper() }}_ --base-iri $(URIBASE)/{{ project.id }} {% endif %}{% endif -%} --print 5 -o $@
+{%- if project.robot_report.upper_ontology is defined and project.robot_report.upper_ontology %}
+
+$(REPORTDIR)/$(SRC)-align-report.tsv: $(SRCMERGED) | $(REPORTDIR) all_robot_plugins
+	$(ROBOT) odk:validate -i $< --reasoner $(REASONER) \
+		 --upper-ontology-iri {{ project.robot_report.upper_ontology }}
```

Comment:
This is a fantastic design decision as it allows us to use version IRIs.

Reply:
It also means that the validation step will download the upper ontology whenever it is needed, which in turns mean that the validation becomes susceptible to network glitches.

I’d like to recommend that projects that want to enable validation against an upper ontology take the supplementary steps of:

* downloading the upper ontology (in the version that they want) once and for all, and commit it somewhere in the repo;
* add an entry in their XML catalog so that the upper ontology IRI redirects to the local file.

Reply:
Does `--upper-ontology-iri` respect the catalogue? That would really be all that is needed to address your concern..

Reply:
Yes it does. I wouldn’t have suggested adding an entry for the upper ontology IRI to the catalog if the `odk:validate` command didn’t use the catalog. ;)